mcs/class/monodoc/Monodoc/providers/EcmaDoc.cs

```diff
@@ -31,7 +31,8 @@ internal class EcmaDoc
 		                                              IDocStorage storage,
 		                                              Dictionary<string, XElement> nsSummaries,
 		                                              Func<XElement, string> indexGenerator = null,
-		                                              IEcmaProviderFileSource fileSource = null)
+		                                              IEcmaProviderFileSource fileSource = null,
```

Comment:
`PopulateTreeFromIndexFile()` taking **8** parameters is a code smell; this is probably a sign that some refactoring is needed, e.g. turning this method into a class and each method parameter into a class property.

It's also odd that EcmaDoc has only static members but is not a static class.

Fortunately, the EcmaDoc type is internal, but still; this is crazy.